NeuralABC_tools/EigenGame_v1.py

```diff
@@ -0,0 +1,40 @@
+import numpy as np
+
+def EigenGame2(data, n):
+    M = np.matmul(data.T, data)
+    dim = M.shape[0]
+    vecs = []
+    gradients = []
+    
+    step = 0.01
+    
+    for i in range(n):
+        vecs.append(np.random.rand(dim))
+    for epoch in range(100000): #number of iterations
+        gradients = []
+        for index in range(len(vecs)):
+            
+            rewards = np.matmul(data, vecs[index])
+            rewards = np.matmul(data.T, data) / dim
```

Comment:
Line 18 overwrites line 17 here. Might need to change the name of one them if they are both used, otherwise remove the unused one.